blocksync/pool.go

```diff
@@ -633,9 +632,26 @@ OUTER_LOOP:
 				}
 			case <-bpr.gotBlockCh:
 				// We got a block!
-				// Continue the for-loop and wait til Quit.
-				continue WAIT_LOOP
+				// Exit the for loop and wait until we receive a quit signal.
+				break WAIT_LOOP
```

Comment:
Will this cause the node to try getting a new peer for the same height in OUTER_LOOP?